compiler/semcall.nim

```diff
@@ -287,7 +287,20 @@ proc getMsgDiagnostic(c: PContext, flags: TExprFlags, n, f: PNode): string =
 
   let ident = considerQuotedIdent(c, f, n).s
   if nfDotField in n.flags and nfExplicitCall notin n.flags:
-    result = errUndeclaredField % ident & result
+    let sym = n.sons[1].typ.sym
+    var typeHint = ""
+    if sym == nil:
+      #[
+      Perhaps we're in a `compiles(foo.bar)` expression, or
+      in a concept, eg:
+        ExplainedConcept {.explain.} = concept o
+          o.foo is int
+      We coudl use: `(c.config $ n.sons[1].info)` to get more context.
```

Comment:
I would prefer normal # comments instead of block comments. But apart from that I think this PR is fine.

Reply:
ok, will fix that in another PR I'll send out shortly (today I hope)

Reply:
=> PR #10239